src/dependencies.jl

```diff
@@ -82,3 +62,49 @@ function get_project_deps(project_file::AbstractString; include_jll::Bool=false)
 
     return project_deps
 end
+
+function clone_all_registries(registry_list::Vector{Pkg.RegistrySpec})
+    registry_temp_dirs = Vector{String}()
+
+    for registry in registry_list
+        tmp_dir = @mock mktempdir(; cleanup=true)
+        local_registry_path = joinpath(tmp_dir, registry.name)
+        push!(registry_temp_dirs, local_registry_path)
+
+        @mock git_clone(registry.url, local_registry_path)
+    end
+
+    return registry_temp_dirs
+end
+
+function get_latest_version_from_registries!(
+    deps::Set{CompatEntry}, registry_list::Vector{Pkg.RegistrySpec}
+)
+    registry_temp_dirs = @mock clone_all_registries(registry_list)
+
+    for registry in registry_temp_dirs
+        registry_toml_path = joinpath(registry, "Registry.toml")
+        registry_toml = TOML.parsefile(joinpath(registry_toml_path))
+        packages = registry_toml["packages"]
+
+        for dep in deps
+            uuid = string(dep.package.uuid)
+
+            if uuid in keys(packages)
+                versions_toml_path = joinpath(
+                    registry, packages[uuid]["path"], "Versions.toml"
+                )
+
+                versions = VersionNumber.(collect(keys(TOML.parsefile(versions_toml_path))))
+                max_version = maximum(versions)
+                dep.version_number = _max(dep.version_number, max_version)
+            end
+        end
+    end
+
+    for tmp_dir in registry_temp_dirs
+        @mock rm(tmp_dir; force=true, recursive=true)
+    end
+
+    return deps
+end
```

Comment:
would maybe something like this make more sense? Seems weird that `get_latest_version_from_registries` should be responsible for removing the tmp dirs. 

This might not be exactly accurate, I forget exactly how all the encapsulation works and if `deps` at the end would actually be mutated

```suggestion
function clone_all_registries(f::Function, registry_list::Vector{Pkg.RegistrySpec})
    registry_temp_dirs = Vector{String}()

    for registry in registry_list
        tmp_dir = @mock mktempdir(; cleanup=true)
        local_registry_path = joinpath(tmp_dir, registry.name)
        push!(registry_temp_dirs, local_registry_path)

        @mock git_clone(registry.url, local_registry_path)
    end

    f(registry_temp_dirs)

    for tmp_dir in registry_temp_dirs
        @mock rm(tmp_dir; force=true, recursive=true)
    end
end

function get_latest_version_from_registries!(
    deps::Set{CompatEntry}, registry_list::Vector{Pkg.RegistrySpec}
)
    @mock clone_all_registries(registry_list) do registry_temp_dirs
        for registry in registry_temp_dirs
            registry_toml_path = joinpath(registry, "Registry.toml")
            registry_toml = TOML.parsefile(joinpath(registry_toml_path))
            packages = registry_toml["packages"]

            for dep in deps
                uuid = string(dep.package.uuid)

                if uuid in keys(packages)
                    versions_toml_path = joinpath(
                        registry, packages[uuid]["path"], "Versions.toml"
                    )

                    versions = VersionNumber.(
                        collect(keys(TOML.parsefile(versions_toml_path)))
                    )
                    max_version = maximum(versions)
                    dep.version_number = _max(dep.version_number, max_version)
                end
            end
        end
    end

    return deps
end
```